drizzlepac/haputils/deconvolve_utils.py

```diff
@@ -131,127 +144,252 @@ def fft_deconv_img(img, psf, freq_limit=0.95, sigma=None):
     # m_map = U (conj(S)/(|S|^2 + alpha^2)) U^H d
     m_map = (D.shape[1] * D.shape[0]) * np.fft.fftshift(np.fft.ifft2(M).real)
 
-    if sigma:
-        m_map = ndimage.gaussian_filter(m_map, sigma=sigma)
+    zero_mask = (img_block > 0).astype(np.int16)
+    m_map *= zero_mask
+
     return m_map
 
-# ======================================================================================================================
 
-# Functions to manage PSF library for deconvolution
+def _center_psf(psf, img_block_shape):
+    """Create centered PSF image with same shape as img_block"""
+    psf_y, psf_x = np.where(psf == psf.max())
+    psf_y = psf_y[0]
+    psf_x = psf_x[0]
+    psf_center = [(img_block_shape[0] // 2), (img_block_shape[1] // 2)]
+    centered_psf = np.zeros(img_block_shape, dtype=psf.dtype)
+
+    # We need to recenter the PSF
+    psf_section = np.where(psf != 0.0)
+    psf_xr = [psf_section[1].min(), psf_section[1].max()]
+    psf_yr = [psf_section[0].min(), psf_section[0].max()]
+    psf_size = [(psf_yr[1] - psf_yr[0]) // 2, (psf_xr[1] - psf_xr[0]) // 2]
+    # If psf_size is not at least 2 * psf_max//2, increase to that value
+    # This will insure that the final position of the PSF in the output is at the exact center
+    psf_max = np.where(psf[psf_yr[0]:psf_yr[1], psf_xr[0]:psf_xr[1]] == psf.max())
+    psf_len = max(max(psf_max[0][0] // 2, psf_size[0]), max(psf_max[1][0] // 2, psf_size[1]))
+
+    centered_psf[psf_center[0] - psf_len: psf_center[0] + psf_len,
+                 psf_center[1] - psf_len: psf_center[1] + psf_len] = psf[psf_y - psf_len: psf_y + psf_len,
+                                                                         psf_x - psf_len: psf_x + psf_len]
+    return centered_psf
+
+
+def pad_arr(arr, block=1024):
+    """ Zero-pad an input array up to an integer number of blocks in each dimension"""
```

Comment:
If this is really a public method, please add description of arguments.

Reply:
Added in d7c635f